Analytical_Engine/Query/Graph/FilterEntities.cs

```diff
@@ -46,6 +46,12 @@ public static partial class Query
         [Output("filter entities", "Entity Dictionary containing the filtered entities.")]
         public static Dictionary<Guid, IBHoMObject> FilterEntities(this Graph graph, Type typeFilter)
         {
+            if (graph == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot filter the entities of a null graph.");
+                return null;
```

Comment:
Why not return an empty collection to follow the pattern?
```suggestion
                return new Dictionary<Guid, IBHoMObject>();
```